Helper/Ratings.php

```diff
@@ -128,14 +132,26 @@ private function getRatingsFromProviders(Product $product, Store $store)
                     return null;
                 }
 
+                $activeProduct = $this->registry->registry('current_product');
+
                 try {
+                    if ($activeProduct !== null) {
+                        $this->registry->unregister('current_product');
+                        $this->registry->register('current_product', $product);
+                    } else  {
+                        $this->registry->register('current_product', $product);
+                    }
+
                     /** @noinspection PhpUndefinedMethodInspection */
                     $ratings = $this->ratingsFactory->create()->getRichSnippet();
                 } catch (\Exception $e) {
+                    $this->registry->register('current_product', $activeProduct);
                     $this->logger->exception($e);
                     return null;
                 }
 
+                $this->registry->register('current_product', $activeProduct);
```

Comment:
This is redundant and it's causing to register twice when running through the indexer, which causes an error.
I'd remove this line.

Reply:
Nopes @supercid, we return in the catch block. If you remove this line we will end up leaving the last product processed by Nosto into the registry